packages/synapse-react-client/src/synapse-client/SynapseClient.ts

```diff
@@ -2074,6 +2075,8 @@ export const uploadFile = (
   contentType: string = file.type,
   progressCallback?: (progress: ProgressCallback) => void,
   getIsCancelled?: () => boolean,
+  onMd5Computed?: () => void,
+  abortController?: AbortController,
```

Comment:
We have the `getIsCancelled` callback, but it's kind of an awkward pattern compared to just passing in an AbortController. I added an ability to pass one but kept `getIsCancelled` to avoid a breaking change.